spec/stack_spec.rb

```diff
@@ -19,56 +17,81 @@
     end
   end
 
-  describe '#pop' do
+  context "when retrieving item" do
     before do
-      stack.push(1)
-      stack.push(9)
-      stack.push(2)
-      stack.push(8)
+      (1 .. 10).each { |num| stack.push(num) }
     end
 
-    it 'retrieves last item and removes it' do
-      expect(stack.size).to eql(4)
+    describe '#pop' do
+      it 'retrieves item and removes it' do
+        expect(stack.size).to eql(10)
 
-      value = stack.pop
-      expect(value).to eql(8)
-      expect(stack.size).to eql(3)
-    end
+        value = stack.pop
+        expect(value).to eql(10)
+        expect(stack.size).to eql(9)
+      end
 
-    # Should not this example raises an error?
-    it "returns 'nil' when stack is empty" do
-      value = stack.pop
+      # Should not this example raises an error?
+      it "returns 'nil' when stack is empty" do
+        stack.clear
 
-      expect(value).to be nil
+        value = stack.pop
+        expect(value).to be_nil
+      end
     end
-  end
 
-  describe '#poll' do
-    it 'retrieves last item without removing it' do
-      expect(stack.size).to eql(4)
+    describe '#poll' do
+      it 'retrieves item without removing it' do
+        expect(stack.size).to eql(10)
 
-      value = stack.poll
+        value = stack.poll
 
-      expect(value).to eql(8)
-      expect(stack.size).to eql(4)
-    end
+        expect(value).to eql(10)
+        expect(stack.size).to eql(10)
+      end
 
-    # Should not this example raises an error?
-    it "returns 'nil' when stack is empty" do
-      expect(stack.poll).to be nil
+      # Should not this example raises an error?
+      it "returns 'nil' when stack is empty" do
+        stack.clear
+        expect(stack.poll).to be_nil
+      end
     end
   end
 
-  describe '#.clear' do
+  describe '#clear' do
     it 'flushes the stack' do
       stack.clear
       expect(stack.size).to eql(0)
     end
   end
 
-  describe '#.size' do
+  describe '#size' do
+    it 'returns 0 when no items' do
+      expect(stack.size).to eql(0)
+    end
+
     it 'returns 3' do
       3.times { stack.push(Object.new) }
+
+      expect(stack.size).to eql(3)
+    end
+  end
+
+  describe '#empty?' do
+    context 'stack contains items' do
+      it "returns 'false'" do
+        3.times { |num| stack.push(num) }
```

Comment:
Pro tip:

You can move this line to a private method inside the spec. At the end of the day a describe block is like a class wrapper. So for example your code here could see like:

```
stack_push(items)
```

And the method:

```
private
def stack_push(items = (1..10))
  items.each {|item| stack.push(item)}
end
```